src/wasm.h

```diff
@@ -1766,18 +1764,48 @@ class StringAs : public SpecificExpression<Expression::StringAsId> {
   void finalize();
 };
 
-class StringViewAccess
-  : public SpecificExpression<Expression::StringViewAccessId> {
+class StringWTF8Advance
+  : public SpecificExpression<Expression::StringWTF8AdvanceId> {
 public:
-  StringViewAccess(MixedArena& allocator) {}
+  StringWTF8Advance(MixedArena& allocator) {}
 
-  StringViewAccessOp op;
+  Expression* ref;
+  Expression* pos;
+  Expression* bytes;
+
+  void finalize();
+};
+
+class StringWTF16Get : public SpecificExpression<Expression::StringWTF16GetId> {
```

Comment:
Nit: I still think `GetCodeUnit` is clearer because everything else matches the instruction name, but I'm not very opinionated. I repost this just in case you dropped it by mistake 😅 In case you intentionally did it nevermind.

Reply:
Oh, sorry, I thought you meant in `wasm-binary.h`. I did make sure to match the binary format there in our own binary constants. But otherwise we have some flexibility I think, both in how we split into classes (and sometimes we invent names for them), and in the names we give. And `StringWTF16GetCodeUnit` is a lot to type :smile: